src/main/java/uk/gov/ons/ctp/sdx/service/impl/ReceiptServiceImpl.java

```diff
@@ -41,6 +41,10 @@ public final void acknowledge(Receipt receipt) throws CTPException {
       caseReceipt.setCaseRef(caseRef.trim());
     }
     caseReceipt.setCaseId((receipt.getCaseId()));
+    String userId = receipt.getUserId();
+    if (userId != null) {
```

Comment:
Will there ever be a time when the user_id going to be blank?

Reply:
If it's always sent through eQ and SDX in the payload I don't think so, it was more precautionary than anything really

Reply:
I think it might be better to assume that it's always going to be there as it's unlikely this'll ever get looked into again.  
As it's currently written SDX will always send it (and it will error and not send you a receipt if it's not there).  I'm not sure if this is correct or not, we should find out as I'd rather nail down how mandatory everything is, as the interface between services doesn't get looked at too often as it's assumed it was correct when it was written (if we imply that it's optional now, it'll always assumed to be optional, regardless of how true that is)

Reply:
After talking with Mebin, sdx-validate says it's mandatory (which informed the decision to make it mandatory in eq) so it'll fall over before it even gets to this gateway.

Make it mandatory, and remove the logic to handle it not being there!